test/RewardsPoolBase.js

```diff
@@ -0,0 +1,181 @@
+		let totalStakedAmount = await RewardsPoolBaseInstance.totalStakedAmount();
+		let reward = blockSinceLastReward.mul(rewardPerBlock)
+		let updatedRewardPerToken = reward.add(rewardPerToken).mul(bOne).div(totalStakedAmount)
+
+		
+		await RewardsPoolBaseInstance.stake(standardStakingAmount);
+
+		let rewardsPerStakedToken = await RewardsPoolBaseInstance.rewardPerStakedToken(0);
+
+		assert(rewardsPerStakedToken.eq(updatedRewardPerToken), "The stake was not successful")
+	})
+
+
+	it.only("Should successfully stake and update the user rewards", async() => {
+		const standardStakingAmount = ethers.utils.parseEther('10') // 10 tokens
+		await stakingTokenInstance.approve(RewardsPoolBaseInstance.contractAddress, ethers.constants.MaxUint256);
+		for (i = 0; i < 50; i++) {
+			//forcing to mine some blocks
+			await stakingTokenInstance.mint(aliceAccount.signer.address,amount);
+        }
+		await RewardsPoolBaseInstance.stake(standardStakingAmount);
+		const blockBeforeStake = await deployer.provider.getBlock('latest');
+		let lastRewardBlock = await RewardsPoolBaseInstance.lastRewardBlock()
+		console.log(lastRewardBlock.toString())
+		let leght2 = await RewardsPoolBaseInstance.getUserRewardDebtLength(aliceAccount.signer.address, 0);
+		let userRewardDebtContract = await RewardsPoolBaseInstance.getUserRewardDebt(aliceAccount.signer.address, 0);
+		
+		let blockSinceLastReward = lastRewardBlock.add(ethers.utils.bigNumberify("1")).sub(ethers.utils.bigNumberify(blockBeforeStake.number))
+		let rewardPerBlock = await RewardsPoolBaseInstance.tokenRewardPerBlock(0)
+		console.log(rewardPerBlock.toString(), "reward per block")
+		let rewardPerToken = await RewardsPoolBaseInstance.rewardPerStakedToken(0)
+		console.log(rewardPerToken.toString(), "reward per token")
+		let totalStakedAmount = await RewardsPoolBaseInstance.totalStakedAmount();
+		console.log(totalStakedAmount.toString(),"total staked amount")
+		
+		for (i = 0; i < 10; i++) {
+			//forcing to mine some blocks
+			await stakingTokenInstance.mint(aliceAccount.signer.address,amount);
+		}
+		lastRewardBlock = await RewardsPoolBaseInstance.lastRewardBlock()
+		let blockSince = blockSinceLastReward.mul(rewardPerBlock)
+		let updatedRewardPerToken = blockSince.add(rewardPerToken).mul(bOne).div(totalStakedAmount)
+		console.log(updatedRewardPerToken.toString(), "updated reward per token")
+		
+		await RewardsPoolBaseInstance.stake(standardStakingAmount);
+		//  rewardPerToken = await RewardsPoolBaseInstance.rewardPerStakedToken(0)
+		//  console.log(rewardPerStakedToken.toString(), "reward per staked tolen after second stake")
+		// let userInfo = await RewardsPoolBaseInstance.userInfo(aliceAccount.signer.address)
+		//  userRewardDebtContract = await RewardsPoolBaseInstance.getUserRewardDebt(aliceAccount.signer.address, 0);
+		// let userTokensOwedContract = await RewardsPoolBaseInstance.getUserOwedLength(aliceAccount.signer.address, 0);
```

Comment:
You are calling the wrong function here and getting the length instead of the amount owed tokens